src/main/java/Deadline.java

```diff
@@ -0,0 +1,57 @@
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+/**
+ * Represents a deadline task that has a specific deadline.
+ * A Deadline object is made up of the task name, as well as a date and time of the deadline.
+ */
+public class Deadline extends Task {
+    protected LocalDateTime by;
```

Comment:
Perhaps by is not the best option to name this variable?

Reply:
Perhaps a more descriptive name such as byDateTime would be better.